tests/Doctrine/Tests/DBAL/Driver/AbstractOracleDriverTest.php

```diff
@@ -44,6 +46,61 @@ public function testReturnsDatabaseNameWithConnectDescriptor()
         self::assertSame($params['user'], $this->driver->getDatabase($connection));
     }
 
+    /**
+     * @param mixed[] $params
+     * @dataProvider easyConnectStringProvider
+     */
+    public function testEasyConnectString(array $params, string $expected) : void
+    {
+        $driver = $this->createMock(AbstractOracleDriver::class);
+
+        $reflection = new ReflectionMethod($driver, 'getEasyConnectString');
+        $reflection->setAccessible(true);
```

Comment:
This is extremely ugly: would it make sense to move this to a value object instead?

Reply:
Having a `ConnectionString` class which implements rendering of `['DESCRIPTION' => ...]` to `(DESCRIPTION=...)` would make sense but it's not what being tested here. I'm testing the logic of mapping of DBAL configuration parameters to the connection string ones which doesn't beling to the `ConnectionString` itself.

Do we need another class which would do the mapping? It will only be used by `AbstractOracleDriver` and will only have one implementation. Besides being able to test it in isolation, what's its purpose?

Reply:
Or… instead of adding the class for mapping, we can create a named constructor like `fromConfigurationParameters()` in the value object and get it tested.